srv/salt/_modules/osd.py

```diff
@@ -1424,16 +1433,23 @@ def __init__(self, osd_id, device, weight, grains, force=False, **kwargs):
         """
         self.osd_id = osd_id
         self.osd_fsid = device.osd_fsid
-        self.partitions = self.set_partitions(device)
+        self.device = device
+        self.partitions = self.set_partitions()
         self._weight = weight
         self._grains = grains
         self.force = force
+        self.keyring = None
+        self.client = None
+        if 'keyring' in kwargs:
```

Comment:
I know we have that in a lot of places. But this notation is easier to read and allows to assign default values:

kwargs.get('keyring', 'a default value')

Reply:
Changing